src/plasma/plasma_protocol.cc

```diff
@@ -0,0 +1,541 @@
+#include "flatbuffers/flatbuffers.h"
+#include "format/plasma_generated.h"
+
+extern "C" {
+#include "plasma_protocol.h"
+
+#include "io.h"
+
+#define FLATBUFFER_BUILDER_DEFAULT_SIZE 1024
+
+/**
+ * Writes an array of object IDs into a vector and return it.
+ *
+ * @param object_ids Array of object IDs to be written.
+ * @param num_objects The number of elements in the array.
+ * @return The string vector containing the object IDs.
+ */
+std::vector<std::string> object_ids_to_vector(object_id object_ids[],
+                                              int64_t num_objects) {
+  std::vector<std::string> result;
+  for (int64_t i = 0; i < num_objects; ++i) {
+    result.push_back(std::string((const char *) &object_ids[i].id[0], UNIQUE_ID_SIZE));
+  }
+  return result;
+}
+
+int plasma_send_CreateRequest(int sock,
+                              protocol_builder *B,
+                              object_id object_id,
+                              int64_t data_size,
+                              int64_t metadata_size) {
+  flatbuffers::FlatBufferBuilder fbb(FLATBUFFER_BUILDER_DEFAULT_SIZE);
+  auto id = fbb.CreateString((char*) &object_id.id[0], UNIQUE_ID_SIZE);
+  auto message = CreatePlasmaCreateRequest(fbb, id, data_size, metadata_size);
+  fbb.Finish(message);
+  return write_message(sock, MessageType_PlasmaCreateRequest, fbb.GetSize(), fbb.GetBufferPointer());
+}
+
+void plasma_read_CreateRequest(uint8_t *data,
+                               object_id *object_id,
+                               int64_t *data_size,
+                               int64_t *metadata_size) {
+  CHECK(data);
+  auto message = flatbuffers::GetRoot<PlasmaCreateRequest>(data);
+  *data_size = message->data_size();
+  *metadata_size = message->metadata_size();
+  CHECK(message->object_id()->size() == UNIQUE_ID_SIZE);
```

Comment:
We probably shouldn't use the `UNIQUE_ID_SIZE` macro, but instead use e.g., `sizeof(*object_id)`.